### PR TITLE
Counting bytes in tcp read write

### PR DESCRIPTION
**Pull request by ErakhtinB**

### CODE REVIEW

File: src/transport/tcp/tcp_connection.cpp [hunk 1]
```diff
@@ -184,6 +184,7 @@ namespace libp2p::transport {
   void TcpConnection::readSome(BytesOut out,
                                size_t bytes,
                                TcpConnection::ReadCallbackFunc cb) {
+    bytes_read_.fetch_add(bytes);
```
Comment (turuslan) [inline src/transport/tcp/tcp_connection.cpp:187]: Should we count bytes we wanted to send, or sent bytes?

File: include/libp2p/transport/tcp/tcp_connection.hpp [hunk 2]
```diff
@@ -115,6 +118,9 @@ namespace libp2p::transport {
 
     std::string debug_str_;
 
+    inline static std::atomic<uint64_t> bytes_read_ = 0;
```
Comment (turuslan) [inline include/libp2p/transport/tcp/tcp_connection.hpp:121]: May move to separate header

**Review by turuslan — APPROVED** [review]

File: include/libp2p/transport/tcp/bytes_counter.hpp [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <atomic>
+
+namespace libp2p::transport {
+
+  class ByteCounter {
```
Comment (turuslan) [inline include/libp2p/transport/tcp/bytes_counter.hpp:7]: May name it `BandwidthMetric`

**Review by turuslan — APPROVED** [review]

File: include/libp2p/transport/tcp/bytes_counter.hpp [hunk 4]
```diff
@@ -0,0 +1,32 @@
+#pragma once
```
Comment (igor-egorov) [inline include/libp2p/transport/tcp/bytes_counter.hpp:1]:
```suggestion
/**
 * Copyright Quadrivium LLC
 * All Rights Reserved
 * SPDX-License-Identifier: Apache-2.0
 */

#pragma once
```

File: src/transport/tcp/bytes_counter.cpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
+#include <libp2p/transport/tcp/bytes_counter.hpp>
```
Comment (igor-egorov) [inline src/transport/tcp/bytes_counter.cpp:1]: Copyright header

File: include/libp2p/transport/tcp/bytes_counter.hpp [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+namespace libp2p::transport {
+
+  class ByteCounter {
+   private:
+    std::atomic<uint64_t> bytes_read_{0};
+    std::atomic<uint64_t> bytes_written_{0};
+
+    ByteCounter() = default;
+
+   public:
```
Comment (igor-egorov) [inline include/libp2p/transport/tcp/bytes_counter.hpp:14]: I was surprised that there is no method to reset the counters. Well, that is another possible approach.

**Review by igor-egorov — APPROVED** [review]
Please check the comments before merge

**Pull request merged**